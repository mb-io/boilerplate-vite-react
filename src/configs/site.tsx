import AboutPage from '@/pages/about/about'
import ContactPage from '@/pages/contact/contact'
import HomePage from '@/pages/home/home'
import NotFoundPage from '@/pages/not-found/not-found'

export type SiteConfig = typeof siteConfig

export const siteConfig = {
  apiBaseURL: 'https://jsonplaceholder.typicode.com',
  name: 'MeteorByte Vite React Boilerplate',
  description:
    'This is a boilerplate build with Vite, React, TypeScript, HeroUI, Zustand, React Router, React Query, Vitest, Testing Library, TailwindCSS, Eslint and Prettier.',
  routes: [
    {
      label: 'Home',
      href: '/',
      component: <HomePage />,
      visibleInNav: true
    },
    {
      label: 'About',
      href: '/about',
      component: <AboutPage />,
      visibleInNav: true
    },
    {
      label: 'Contact',
      href: '/contact',
      component: <ContactPage />,
      visibleInNav: true
    },
    {
      label: 'Page not found',
      href: '/*',
      component: <NotFoundPage />,
      visibleInNav: false
    }
  ],
  links: {
    github: 'https://github.com/mb-io/boilerplate-vite-react'
  }
}
