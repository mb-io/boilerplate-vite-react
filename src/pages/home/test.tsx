import { render, screen } from '@testing-library/react'

import HomePage from './home'
import { siteConfig } from '@/configs/site'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

const queryClient = new QueryClient()

describe('<HomePage />', () => {
  it('should render the HomePage', () => {
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <HomePage />
      </QueryClientProvider>
    )

    expect(
      screen.getByRole('heading', {
        name: /Welcome!/i,
        level: 1
      })
    ).toBeInTheDocument()

    expect(screen.getByText(siteConfig.description)).toBeInTheDocument()

    expect(
      screen.getByRole('link', {
        name: /Open in VS Code/i
      })
    ).toBeInTheDocument()

    expect(screen.getByRole('img')).toBeInTheDocument()

    expect(container.firstChild).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
