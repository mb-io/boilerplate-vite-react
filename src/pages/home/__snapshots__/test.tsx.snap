// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<HomePage /> > should render the HomePage 1`] = `
<div>
  <div
    class="relative overflow-hidden bg-white dark:bg-black"
  >
    <div
      class="h-screen sm:pb-40 sm:pt-24 lg:pb-48 lg:pt-40"
    >
      <div
        class="relative mx-auto max-w-7xl px-4 sm:static sm:px-6 lg:px-8"
      >
        <div
          class="sm:max-w-lg"
        >
          <div
            class="my-4"
          >
            <img
              class="inline-block rounded-full w-14 h-14"
              src="data:image/svg+xml,%3csvg%20width='410'%20height='410'%20viewBox='0%200%20410%20410'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M205%20410C318.218%20410%20410%20318.218%20410%20205C410%2091.7816%20318.218%200%20205%200C91.7816%200%200%2091.7816%200%20205C0%20318.218%2091.7816%20410%20205%20410ZM205%20360C290.604%20360%20360%20290.604%20360%20205C360%20119.396%20290.604%2050%20205%2050C119.396%2050%2050%20119.396%2050%20205C50%20290.604%20119.396%20360%20205%20360Z'%20fill='%234F46E5'%20/%3e%3c/svg%3e"
            />
          </div>
          <h1
            class="text-4xl font-bold tracking-tight text-gray-900 sm:text-6xl"
          >
            Welcome!
          </h1>
          <p
            class="mt-4 text-xl text-gray-500"
          >
            This is a boilerplate build with Vite, React, TypeScript, HeroUI, Zustand, React Router, React Query, Vitest, Testing Library, TailwindCSS, Eslint and Prettier.
          </p>
          <p
            class="mt-4 text-xl text-gray-500"
          />
        </div>
        <div
          class="my-10"
        >
          <a
            class="inline-block rounded-md border border-transparent bg-indigo-600 px-8 py-3 text-center font-medium text-white hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-700 focus:ring-offset-2"
            href="vscode://"
          >
            Open in VS Code
          </a>
          <div
            aria-hidden="true"
            class="pointer-events-none mt-10 md:mt-0 lg:absolute lg:inset-y-0 lg:mx-auto lg:w-full lg:max-w-7xl"
          >
            <div
              class="absolute sm:left-1/2 sm:top-0 sm:translate-x-8 lg:left-1/2 lg:top-1/2 lg:-translate-y-1/2 lg:translate-x-8"
            >
              <div
                class="flex items-center space-x-2"
              >
                <div
                  class="grid shrink-0 grid-cols-1 gap-y-2"
                >
                  <div
                    class="h-64 w-44 overflow-hidden rounded-lg sm:opacity-0 lg:opacity-100"
                  >
                    <img
                      alt=""
                      class="size-full bg-indigo-100 object-cover object-center"
                      src="https://picsum.photos/600?random=1"
                    />
                  </div>
                  <div
                    class="h-64 w-44 overflow-hidden rounded-lg sm:opacity-0 lg:opacity-100"
                  >
                    <img
                      alt=""
                      class="size-full bg-indigo-100 object-cover object-center"
                      src="https://picsum.photos/600?random=2"
                    />
                  </div>
                </div>
                <div
                  class="grid shrink-0 grid-cols-1 gap-y-2"
                >
                  <div
                    class="h-64 w-44 overflow-hidden rounded-lg sm:opacity-0 lg:opacity-100"
                  >
                    <img
                      alt=""
                      class="size-full bg-indigo-100 object-cover object-center"
                      src="https://picsum.photos/600?random=3"
                    />
                  </div>
                  <div
                    class="h-64 w-44 overflow-hidden rounded-lg sm:opacity-0 lg:opacity-100"
                  >
                    <img
                      alt=""
                      class="size-full bg-indigo-100 object-cover object-center"
                      src="https://picsum.photos/600?random=4"
                    />
                  </div>
                  <div
                    class="h-64 w-44 overflow-hidden rounded-lg sm:opacity-0 lg:opacity-100"
                  >
                    <img
                      alt=""
                      class="size-full bg-indigo-100 object-cover object-center"
                      src="https://picsum.photos/600?random=5"
                    />
                  </div>
                </div>
                <div
                  class="grid shrink-0 grid-cols-1 gap-y-2"
                >
                  <div
                    class="h-64 w-44 overflow-hidden rounded-lg sm:opacity-0 lg:opacity-100"
                  >
                    <img
                      alt=""
                      class="size-full bg-indigo-100 object-cover object-center"
                      src="https://picsum.photos/600?random=6"
                    />
                  </div>
                  <div
                    class="h-64 w-44 overflow-hidden rounded-lg sm:opacity-0 lg:opacity-100"
                  >
                    <img
                      alt=""
                      class="size-full bg-indigo-100 object-cover object-center"
                      src="https://picsum.photos/600?random=7"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
